packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -930,12 +930,16 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
         widget.onInteractionEnd?.call(ScaleEndDetails());
         return;
       }
-      final RenderBox childRenderBox = _childKey.currentContext!.findRenderObject()! as RenderBox;
-      final Size childSize = childRenderBox.size;
-      final double scaleChange = 1.0 - event.scrollDelta.dy / childSize.height;
-      if (scaleChange == 0.0) {
+      
+      // Ignore left and right scroll.
+      if (event.scrollDelta.dy == 0.0) {
         return;
       }
+
+      // The mouseWheel scroll amount is hardcoded to 20 while a trackpad scroll can be any amount
+      // by dividing the scroll amount by 200 it scale 0.1 units per mouseWheels scroll and feel natural for trackpads.
+      // See https://github.com/flutter/flutter/pull/71266
+      final double scaleChange = math.exp(-event.scrollDelta.dy / 200);
```

Comment:
Good idea using e.  I tried this again on my Macbook's trackpad and also using a Magic Mouse, and both felt natural to me.